cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -166,12 +166,9 @@ static void sc_populate_libgl_with_hostfs_symlinks(const char *libgl_dir,
 		char prefix_dir[512] = { 0 };
 		const char *pathname = glob_res.gl_pathv[i];
 		char *pathname_copy1
-		    SC_CLEANUP(sc_cleanup_string) = strdup(pathname);
+		    SC_CLEANUP(sc_cleanup_string) = sc_strdup(pathname);
```

Comment:
Wondering if maybe `strndup()` wrapper would be useful too. The upper bound of size is usually known, and the implementations I've seen all used `strnlen()` so as not to allocate the max size unnecessarily.

Reply:
I'm happy to add one once we use strndup more than once.